src/main/webapp/app/overview/course-dashboard/course-exercise-performance/course-exercise-performance.component.ts

```diff
@@ -0,0 +1,104 @@
+        selectable: true,
+        group: ScaleType.Ordinal,
+        domain: [YOUR_GRAPH_COLOR, AVERAGE_GRAPH_COLOR],
+    };
+    yScaleMax = 100;
+
+    protected readonly round = round;
+    protected readonly Math = Math;
+    protected readonly YOUR_GRAPH_COLOR = YOUR_GRAPH_COLOR;
+    protected readonly AVERAGE_GRAPH_COLOR = AVERAGE_GRAPH_COLOR;
+
+    constructor(private translateService: TranslateService) {
+        this.translateService.onLangChange.subscribe(() => {
+            this.setupChart();
+        });
+    }
+
+    ngOnInit(): void {
+        this.setupChart();
+    }
+
+    ngOnChanges(): void {
+        this.setupChart();
+    }
+
+    /**
+     * This getter checks if there is data available for the chart.
+     * It checks if `ngxData` is defined, if it has at least one entry, and if at least one of those entries has a non-empty `series` array.
+     * @returns {boolean} - Returns true if data is available for the chart, false otherwise.
+     */
+    get isDataAvailable(): boolean {
+        return this.ngxData && this.ngxData.length > 0 && this.ngxData.some((data) => data.series.length > 0);
+    }
+
+    /**
+     * This method is responsible for setting up the chart that displays the performance of the exercises.
+     * It translates the labels for the chart, prepares the data for the chart, and calculates the maximum value for the y-axis.
+     */
+    private setupChart(): void {
+        this.yourScoreLabel = this.translateService.instant('artemisApp.courseStudentDashboard.exercisePerformance.yourScoreLabel');
+        this.averageScoreLabel = this.translateService.instant('artemisApp.courseStudentDashboard.exercisePerformance.averageScoreLabel');
+
+        this.ngxData = [
+            {
+                name: this.yourScoreLabel,
+                series: this.exercisePerformance.map((performance) => {
+                    return {
+                        name: performance.shortName?.toUpperCase() || performance.title,
+                        value: performance.score || 0, // If the score is undefined, set it to 0
+                        extra: {
+                            title: performance.title,
+                        },
+                    };
+                }),
+            },
+            {
+                name: this.averageScoreLabel,
+                series: this.exercisePerformance.map((performance) => {
+                    return {
+                        name: performance.shortName?.toUpperCase() || performance.title,
+                        value: performance.averageScore || 0,
+                        extra: {
+                            title: performance.title,
+                        },
+                    };
+                }),
+            },
+        ];
+
+        const maxScore = Math.max(...this.ngxData.flatMap((data) => data.series.map((series) => series.value)));
+        this.yScaleMax = Math.max(100, Math.ceil(maxScore / 10) * 10);
```

Comment:
I think that this is really hard to read, would you mind introducing a method for this?

```suggestion
        this.yScaleMax = Math.max(100, Math.ceil(maxScore / 10) * 10);
```

a good method name might be `roundUpToNextMultipleOfTen`?